source/grid/grid_in.cc

```diff
@@ -2542,6 +2566,162 @@ void GridIn<dim, spacedim>::read_tecplot(std::istream &)
 }
 
 
+
+template <int dim, int spacedim>
+void GridIn<dim, spacedim>::read_assimp(const std::string &filename,
+                                        const unsigned int mesh_index,
+                                        const bool remove_duplicates,
+                                        const double tol,
+                                        const bool ignore_unsupported_types)
+{
+#ifdef DEAL_II_WITH_ASSIMP
+  // Only good for surface grids.
+  AssertThrow(dim<3, ExcImpossibleInDim(dim));
+
+  // Create an istance of the Importer class
+  Assimp::Importer importer;
+
+  // And have it read the given file with some  postprocessing
+  const aiScene *scene = importer.ReadFile( filename.c_str(),
+                                            aiProcess_RemoveComponent   |
+                                            aiProcess_JoinIdenticalVertices   |
+                                            aiProcess_ImproveCacheLocality  |
+                                            aiProcess_SortByPType   |
+                                            aiProcess_OptimizeGraph   |
+                                            aiProcess_OptimizeMeshes);
+
+  // If the import failed, report it
+  AssertThrow(scene, ExcMessage(importer.GetErrorString()))
+
+  AssertThrow(scene->mNumMeshes, ExcMessage("Input file contains no meshes."));
+
+  AssertThrow((mesh_index == numbers::invalid_unsigned_int) ||
+              (mesh_index < scene->mNumMeshes),
+              ExcMessage("Too few meshes in the file."));
+
+  int start_mesh = (mesh_index == numbers::invalid_unsigned_int ?
+                    0 : mesh_index);
```

Comment:
since `mesh_index` is unsigned, you probably want this variable (and the next) to be unsigned as well